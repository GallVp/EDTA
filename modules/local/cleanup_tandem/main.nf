process CLEANUP_TANDEM {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/64/64d26063bedc2efcba20750a408a21f50907986d0d9aee685b03d7d05d3fbd8b/data':
        'community.wave.seqera.io/library/trf_perl:44f9844b1bf765c3' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.fasta")    , emit: cleaned_fasta
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix  = task.ext.prefix ?: "${meta.id}"
    def args    = task.ext.args ?: ''
    if ( "$fasta" == "${prefix}.fasta" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    cleanup_tandem.pl \\
        -f $fasta \\
        $args \\
        > ${prefix}.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl -v | sed -n 's|This is perl.*(\\(.*\\)).*|\\1|p')
        trf: \$(trf -v |& sed -n 's|.*Version \\(.*\\)|\\1|p')
    END_VERSIONS
    """

    stub:
    def prefix  = task.ext.prefix ?: "${meta.id}"
    if ( "$fasta" == "${prefix}.fasta" ) error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    touch ${prefix}.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl -v | sed -n 's|This is perl.*(\\(.*\\)).*|\\1|p')
        trf: \$(trf -v |& sed -n 's|.*Version \\(.*\\)|\\1|p')
    END_VERSIONS
    """
}