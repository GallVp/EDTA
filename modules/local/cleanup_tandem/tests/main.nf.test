nextflow_process {

    name "Test Process CLEANUP_TANDEM"
    script "../main.nf"
    process "CLEANUP_TANDEM"
    
    tag "cleanup_tandem"
    tag "modules_local"
    tag "modules"

    test("sarscov2 - genome") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'sarscov2' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - genome - stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id: 'sarscov2' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

}
