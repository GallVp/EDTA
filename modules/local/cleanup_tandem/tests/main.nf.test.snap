{
    "sarscov2 - genome - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "sarscov2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2cce04ed211ff9c6f7f322d2b217af83"
                ],
                "cleaned_fasta": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "sarscov2.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2cce04ed211ff9c6f7f322d2b217af83"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-28T16:12:53.161124"
    },
    "sarscov2 - genome": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "sarscov2.fasta:md5,e73599798195a519ba2565c3f0275b93"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2cce04ed211ff9c6f7f322d2b217af83"
                ],
                "cleaned_fasta": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "sarscov2.fasta:md5,e73599798195a519ba2565c3f0275b93"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2cce04ed211ff9c6f7f322d2b217af83"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-28T16:12:48.160979"
    }
}